segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/tables/TableStore.java

```diff
@@ -144,6 +145,32 @@ default int maximumValueLength() {
      */
     CompletableFuture<List<Long>> put(String segmentName, List<TableEntry> entries, Duration timeout);
 
+    /**
+     * Inserts new or updates existing Table Entries into the given Table Segment.
```

Comment:
add " conditioned on the Segment's Length matching an expected value".

do this for `remove` too.